processing/src/main/java/org/apache/druid/segment/serde/DictionaryEncodedColumnPartSerde.java

```diff
@@ -305,6 +311,33 @@ public void read(ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnCo
 
         builder.setType(ValueType.STRING);
 
+        final int dictionaryStartPosition = buffer.position();
+        final byte dictionaryVersion = buffer.get();
+
+        if (dictionaryVersion == EncodedStringDictionaryWriter.VERSION) {
+          final byte encodingId = buffer.get();
+          if (encodingId == StringEncodingStrategy.FRONT_CODED_ID) {
+            readFrontCodedColumn(buffer, builder, rVersion, rFlags, hasMultipleValues);
+          } else {
+            throw new ISE("impossible, unknown encoding strategy id: %s", encodingId);
```

Comment:
We should likely also support an ID for the GenericIndexed based approach so that in a future release we can protect from GenericIndexed getting a new version number by persisting those columns as just another encoding here.